packages/liveblocks-core/src/lib/Poller.ts

```diff
@@ -23,32 +32,35 @@ export function makePoller(
 
   function poll() {
     if (context.state === "running") {
-      //
-      // XXX Should we make this contingent on the callback's execution, i.e. do
-      // XXX try { await callback() } finally { schedule() } instead, so that
-      // XXX two polling functions will never "overlap"?
+      // XXX Set a max timeout for the `callback()` (make `callback` take
+      // a signal, and protect each call with AbortSignal.timeout)
```

Comment:
Yes! This is a must-have before we merge this change in, and we should unit test it properly.